modules/git/repo.go

```diff
@@ -283,7 +283,7 @@ type DivergeObject struct {
 // GetDivergingCommits returns the number of commits a targetBranch is ahead or behind a baseBranch
 func GetDivergingCommits(ctx context.Context, repoPath, baseBranch, targetBranch string) (do DivergeObject, err error) {
 	cmd := NewCommand(ctx, "rev-list", "--count", "--left-right").
-		AddDynamicArguments(baseBranch + "..." + targetBranch)
+		AddDynamicArguments(baseBranch + "..." + targetBranch).AddArguments("--")
```

Comment:
Sorry, meant this change.
(Did GitHub change its mechanism what is displayed when?)

Reply:
And `GetDivergingCommits` is covered by tests, adding `--` is right here.

Reply:
Perhaps we should add a comment for future readers:
```suggestion
		AddDynamicArguments(baseBranch + "..." + targetBranch).AddArguments("--") // '--' needed so that git knows what is a reference and what is a file in case of an error 
```

Reply:
This is not the only place using it. I don't think it's worth to repeat such comments. It is a git's standard mechanism.


![image](https://github.com/go-gitea/gitea/assets/2114189/bb033a19-ba88-4606-b911-df175050c984)